ir_datasets/datamaestro/data.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from typing import Iterator
+from datamaestro_text.interfaces.trec import parse_qrels
+from experimaestro import Config, Option
+import datamaestro_text.data.ir as ir
+import ir_datasets
+
+# Interface between ir_datasets and datamaestro:
+# provides adapted data types
+
+
+class IRDSId(Config):
+    @classmethod
+    def __xpmid__(cls):
+        return f"ir_datasets.{cls.__qualname__}"
+
+    irds: Option[str]
+
+
+class AdhocTopics(ir.AdhocTopics, IRDSId):
+    def iter(self) -> Iterator[ir.AdhocTopic]:
+        """Returns an iterator over topics"""
+        ds = ir_datasets.load(self.irds)
+        for query in ds.queries_iter():
+            yield ir.AdhocTopic(query.query_id, query.text)
+
+
+class AdhocAssessments(ir.AdhocAssessments, IRDSId):
+    def iter(self):
+        """Returns an iterator over assessments"""
+        ds = ir_datasets.load(self.irds)
+        yield from parse_qrels(Path(ds.qrels_path()))
```

Comment:
Not all datasets provide qrels in the TREC format. I recommend using `qrels_iter` instead.

Reply:
Yes this was planned - it was just quicker to start with.